packages/snjs/lib/services/api/api_service.ts

```diff
@@ -188,13 +198,8 @@ export class SNApiService extends PureService {
     }
   }
 
-  private async request(params: {
-    verb: HttpVerb;
-    url: string;
-    fallbackErrorMessage: string;
-    params?: HttpParams;
-    authentication?: string;
-  }) {
+  private async request(params: HttpRequestWithFallbackError):
+  Promise<HttpResponse<unknown>> {
```

Comment:
Do you think it's better to have the response content be `unknown` than, say, a generic type?
```suggestion
  private async request<T>(params: HttpRequestWithFallbackError):
  Promise<HttpResponse<T>> {
```

On one hand I think `unknown` is more correct, on the other casting everywhere always seemed a little dangerous to me. Though that may not be a rational feeling.

Reply:
Definitely `unknown` should be avoided as much as possible. Casting should be avoided even more. That's a sensible feeling.

Used unknown here to be consistent with what `this.httpService.runHttp` and `this.errorResponseWithFallbackMessage` return.

Fixing that will require quite a bit of refactoring (I actually started it and then backed out recognizing the extent) -- outside of scope of this PR.

I did create a task for this however: https://app.asana.com/0/1185703377033953/1200169272405083/f

Reply:
Cool, thank you for expanding your reasoning for me 🙂